src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -620,6 +620,11 @@ public virtual NamedTypeSymbol TupleUnderlyingType
         public virtual ImmutableArray<string> TupleElementNames => default(ImmutableArray<string>);
```

Comment:
This is an internal API that exposes readily available data that don't have to be built. If one needs names, this is the cheapest, most efficient API to get them. 

---
In reply to: [64280967](https://github.com/dotnet/roslyn/pull/11457#discussion_r64280967) [](ancestors = 64280967)